scripts/globals/magic.lua

```diff
@@ -711,19 +714,22 @@ function calculateMagicBurst(caster, spell, target)
             -- Something strange is going on if this occurs.
             burst = 1.0;
         end
+    end
 
-        --add burst bonus for BLM AMII spells
-        if (spell:getID() == 205 or spell:getID() == 207 or spell:getID() == 209 or spell:getID() == 211 or spell:getID() == 213 or spell:getID() == 215) then
-            if (caster:getMerit(blmAMIIMerit[spell:getElement()]) ~= 0) then -- no bonus if the caster has zero merit investment - don't want to give them a negative bonus
-                burst = burst + (caster:getMerit(blmAMIIMerit[spell:getElement()]) - 1) * 0.03; -- bonus value granted by merit is 1; subtract 1 since unlock doesn't give a magic burst bonus
-                -- print((caster:getMerit(blmAMIIMerit[spell:getElement()]) - 1) * 0.03)
-            end
+    --obtain the magic burst bonus from job traits and equipment, as well as BLM merits for AMII spells
+
+    magicburstbonus = magicburstbonus + (caster:getMod(MOD_MAG_BURST_BONUS) / 100);
+
+    --add burst bonus for BLM AMII spells
+    if (spell:getID() == 205 or spell:getID() == 207 or spell:getID() == 209 or spell:getID() == 211 or spell:getID() == 213 or spell:getID() == 215) then
```

Comment:
no, I'd rather see it be a generic magic burst bonus parameter that the spell script passes in.  The spells themselves already know they should have a magic burst, there's no reason for magic.lua to go and figure that out again.

@takhlaq can you see if it's feasible to mass replace the calls to calculateMagicDamage to use a single table for all the params? like we did for WS forever ago

Reply:
maybe just at top area of magic global instead of status.lua because status is getting to be a bloaty fat global these days, and I don't think anything else is gonna use that table

Reply:
> @takhlaq can you see if it's feasible to mass replace the calls to calculateMagicDamage to use a single table for all the params? like we did for WS forever ago

https://github.com/DarkstarProject/darkstar/commit/ef05e1b046d824a0a1a1da413a48e2b15109eeda

Reply:
perfect!
@Whitechaser can you pull those changes and then add the magic burst bonus as params?

Reply:
Hmmhmm. Okay, I sorta get what to do for the AM spell scripts. Do you want me to do reproduce those changes to the rest of the spell scripts that weren't edited in that commit?